### PR TITLE
Added metadata handlers for HTTP API

### PR DESCRIPTION
**Pull request by commx**
This pull request replaces #4080.

I've also added `LUA-AXFR-SCRIPT` to the list of metadata entries which cannot be modified via API due to security concerns. As discussed previously, the implementation will reject creation and modification of custom metadata entries. We may choose from three possible options to relax the behavior:

1. Leave the implementation as-is, rejecting custom metadata entries at all
2. Accept custom metadata if the configuration option `api-allow-custom-metadata` is set to `yes`, reject otherwise
3. Accept custom metadata at all

### CODE REVIEW

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -526,6 +526,181 @@ static void updateDomainSettingsFromDocument(const DomainInfo& di, const DNSName
...
+      throw ApiException("Could not update metadata entries for domain '" + zonename.toString() + "'");
+
+    Json::object key {
+      { "type", "Metadata" },
+      { "kind", document["kind"] },
+      { "metadata", metadata }
+    };
+
+    resp->status = 201;
+    resp->setBody(key);
```
Comment (zeha) [inline pdns/ws-auth.cc]: If you return 201 `No Content`, there should be no body really 😄

**Comment by zeha** [comment]
Small nit, can't find anything to complain about otherwise. </austrian>

Looks good!

**Comment by zeha** [comment]
Oh, updating the docs would be heart-warming.

File: pdns/ws-auth.cc (hunk 1 above)
Comment (commx) [inline pdns/ws-auth.cc]: From the API spec about this:
> POST: Create. Success reply: 201 Created, with new object as body.

Comment (zeha) [inline pdns/ws-auth.cc]: You are, once again, correct. (And I fail at remembering the spec.)

**Comment by zeha** [comment]
IMO good to merge; if I was wrong with rgd. to response codes, this can be fixed afterwards.

**Pull request merged**